### PR TITLE
Upgraded scalatest, scalactic and scalacheck to the versions 3.2.10, 3.2.10 and 1.15 respectively

### LINKED ISSUE #3719: Introduce support for effectful testing
## Overview

Currently, we don't have support for `F[_]` effectful computation in tests so tests are using `_.runSyncUnsafe()` to get the value and assert result. This is not desired, recommended, runs slower and also more more difficult to read and write.

## Solution

Introduce libraries to extend _Scalatest_ and _scalacheck_ to support working with `F[_]` values without hacks.
https://typelevel.org/cats-effect/docs/2.x/guides/testing

Two libraries that we need are:
1. `cats-effect-testing` which enables body of tests to be effect type like IO or Task
  https://github.com/typelevel/cats-effect-testing#scalatest
1. `ScalaCheck Effect` which enables property testing with effectful values `forAllF`
  https://github.com/typelevel/scalacheck-effect#scalacheck-effect

- [x] Upgrade Scalatest, scalacheck and scalactic to latest versions to support cats-effect-testing which can only works with Scalatest 3.1.0
  https://github.com/typelevel/cats-effect-testing/blob/v0.3.0/build.sbt#L56
- [x] Introduce `monix-testing-scalatest` library to support minix Task in tests
  https://github.com/monix/monix-testing
- [ ] Introduce `cats-effect-testing` and `scalacheck-effect` into the project and improve InMemoryKeyValueStoreSpec and LmdbKeyValueStoreSpec by using effectful version. This means removal of _.runSyncUnsafe() by using `forAllF` instead of `forAll` together with using Prop instead of assert or should (e.g. `shouldBe` is the same as `?=`).
    - examples of property based tests can be found here (for now only on feature/v0.12 branch)
     https://github.com/rchain/rchain/blob/40d15bf88d343ea5eea904a6bfc82306ce696615/casper/src/test/scala/coop/rchain/casper/util/BlockHashSignatureSpec.scala#L14-L72
- [ ] Add WebApi tests as property based tests like for KeyValueStore. Generate arbitrary input and output data and assert ib underlying BlockApi received right input and returned right output.

### PR DESCRIPTION
**Pull request by stanislavlyalin**
## Overview

Upgraded version of [scalatest, scalactic](https://www.scalatest.org/) and [scalacheck](https://scalacheck.org/) libraries to be able to use effectful unit testing.

The libraries have not been updated to the latest versions, but to
* 3.2.10 for scalatest/scalactic and
* 1.15 for scalacheck
to provide compatibility with [cats-effect-testing](https://github.com/typelevel/cats-effect-testing) (see [release 1.4.0 notes](https://github.com/typelevel/cats-effect-testing/releases/tag/v1.4.0) and [PR](https://github.com/typelevel/cats-effect-testing/pull/208)).

### Note

* Changing imports made according to [scalatest 3.1.0 release notes](https://www.scalatest.org/release_notes/3.1.0).
* Renaming of scalacheck entities made according to [scalatest autofix warnings](https://github.com/scalatest/autofix/blob/d1fad4842e65c9c54e710d8e9b9de39dfe4527ae/3.0.x/rules/src/main/scala/org/scalatest/autofix/v3_0_x/RenameDeprecatedPackage.scala).

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: comm/src/test/scala/coop/rchain/comm/rp/ConnectSpec.scala [hunk 1]
```diff
@@ -15,7 +15,9 @@ import org.scalatest._
 
 import scala.concurrent.duration.{FiniteDuration, MILLISECONDS}
 
-class ConnectSpec extends FunSpec with Matchers with BeforeAndAfterEach with AppendedClues {
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+class ConnectSpec extends AnyFunSpec with Matchers with BeforeAndAfterEach with AppendedClues {
```
Comment (tgrospic) [inline comm/src/test/scala/coop/rchain/comm/rp/ConnectSpec.scala:19]: Please optimize and sort imports with empty line before class definition.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -72,9 +72,9 @@ object Dependencies {
   val pureconfig          = "com.github.pureconfig"      %% "pureconfig"                % "0.14.0"
   val scalaLogging        = "com.typesafe.scala-logging" %% "scala-logging"             % "3.9.4"
   val scalaUri            = "io.lemonlabs"               %% "scala-uri"                 % "3.0.0"
-  val scalacheck          = "org.scalacheck"             %% "scalacheck"                % "1.15.2"
-  val scalacheckShapeless = "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % "1.2.5" % "test"
-  val scalactic           = "org.scalactic"              %% "scalactic"                 % "3.0.5" % "test"
+  val scalacheck          = "org.scalacheck"             %% "scalacheck"                % "1.15.0"
+  val scalacheckShapeless = "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % "1.2.5"  % "test"
```
Comment (tgrospic) [inline project/Dependencies.scala]: Should _scalacheck-shapeless_ be updated to `scalacheck-shapeless_1.15` as well?

File: casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala [hunk 3]
```diff
@@ -39,7 +41,7 @@ class BondedStatusAPITest
       blockApi <- createBlockApi(node)
       res <- blockApi
               .bondStatus(ByteString.copyFrom(publicKey.bytes), block.some)
-              .map(_.right.value)
+              .map(_.value)
```
Comment (tgrospic) [inline casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala:44]: Is this change somehow related to this PR?

File: casper/src/test/scala/coop/rchain/casper/api/ExploratoryDeployAPITest.scala [hunk 4]
```diff
@@ -68,7 +70,7 @@ class ExploratoryDeployAPITest
                      "new return in { for (@data <- @\"store\") {return!(data)}}",
                      b2.blockHash
                    )
-          (par, b) = result.right.value
+          (par, b) = result.value
```
Comment (tgrospic) [inline casper/src/test/scala/coop/rchain/casper/api/ExploratoryDeployAPITest.scala:73]: This also?

File: casper/src/test/scala/coop/rchain/casper/api/LastFinalizedAPITest.scala [hunk 5]
```diff
@@ -30,7 +32,7 @@ class LastFinalizedAPITest
     for {
       blockApi <- createBlockApi(node)
       res      <- blockApi.isFinalized(block.blockHash.toHexString)
-    } yield res.right.value
+    } yield res.value
```
Comment (tgrospic) [inline casper/src/test/scala/coop/rchain/casper/api/LastFinalizedAPITest.scala:35]: And this?

File: casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala (hunk 3 above)
Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala:44]: Oh, I see now this is an improvement in the new version of ScataTest to extract value from Either with test failed error.

File: comm/src/test/scala/coop/rchain/comm/rp/ConnectSpec.scala (hunk 1 above)
Comment (stanislavlyalin) [reply comm/src/test/scala/coop/rchain/comm/rp/ConnectSpec.scala:19]: [Done](https://github.com/rchain/rchain/pull/3721/commits/580a39185ca80361b7c8fb6f6a7435d90c1536a8).

File: project/Dependencies.scala (hunk 2 above)
Comment (stanislavlyalin) [reply project/Dependencies.scala]: Yes, you are right. On the [releases](https://github.com/alexarchambault/scalacheck-shapeless/releases) page I see that scalacheck 1.15 used from 1.3.0 version (the latest release for now). I [upgraded](https://github.com/rchain/rchain/pull/3721/commits/796307f35078eb50c2977db7a5a5a1c9ba456d9e) scalacheck-shapeless version.

**Review by tgrospic — APPROVED** [review]
Excellent!

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by stanislavlyalin** [comment]
bors merge
